api-call.py

```diff
@@ -1,29 +1,334 @@
+        )
 
-def format_result(data):
-    text = json.dumps(data, sort_keys=True, indent=4)
-    print(text)
+        logging.info(f'{current_time}: {message}')
 
 
+    def get_access_token(self) -> str:
+        """Get access token from OAuth2 for connecting with OSU public APIs
 
+        :returns: The access token string
+        """
+        url = self.config['access_token']['url']
+        payload = self.config['access_token']['payload']
+        payload['grant_type'] = 'client_credentials'
+        headers = {'Content-Type': 'application/x-www-form-urlencoded;charset=utf-8'}
 
-if __name__ == "__main__":
-    parameters = {
-            
-        }
+        response = requests.post(
+            url,
+            data=payload,
+            headers=headers,
+            timeout=10
+        )
 
-    header={"Content-Type": "application/x-www-form-urlencoded;charset=utf-8" , "Authorization":"Bearer CUTOuxWmuRfKQKynBGjFGmBROM9R"}
+        if response.status_code == 200:
+            access_token = response.json()['access_token']
+            access_token_extended = f'Bearer {access_token}'
+            return access_token_extended
+        else:
+            self.log_message(f'Request failed with status code: {response.status_code}')
+            return ''
+
+
+    def get_api_data(self, url: str, parameters: Dict[str, Any], header: Dict[str, str]) -> Any:
+        """Get Request for getting the data from API
+
+        :param url: URL of the API
+        :param parameters: The parameters for API
+        :param header: The headers for API with authorization
+        :return: The response data from the API
+        """
+
+        response = requests.get(
+            url,
+            params=parameters,
+            headers=header,
+            timeout=10
+        )
+        if response.status_code == 200:
+            response_data = self.format_result(response)
+            return response_data
+        else:
+            print(response.status_code)
+            generated_error = f'There is a {response.status_code} error with this request'
+            self.log_message(generated_error)
+
+
+    def format_result(self, response: requests.Response) -> Any:
+        """Returns data from response object after JSON validation
+
+        :param response: The API response object
+        :return: The validated data
+        """
+        try:
+            data = response.json()
+        except json.decoder.JSONDecodeError:
+            print("Invalid JSON response")
+
+        text = json.dumps(data, indent=4)
+        return data
+
+
+    def show_tasks(self) -> None:
+        """Show tasks to choose as a Menu
+        """
+        api_options = ['Quit',
+                    'Beavers Bus',
+                    'Terms',
+                    'Search Text Books of a Term by a custom/any Date',
+                    'Get Details of a Vehicle and its Stops given a specified Route, with ETA and current destination']
```

Comment:
The indent formatting here also violates the styling guild. Make sure you format them correctly and use a linter to ensure there is no error/warning.

Reply:
done